### PR TITLE
SDK support for hbase snapshot & restore commands

### PR DESCRIPTION
**Pull request by adeshr**

### CODE REVIEW

**Comment by mindprince** [comment]
Fix failing tests.

**Comment by vrajat** [comment]
Looks good to me. Caveat: Integration tests are successful.

File: tests/test_cluster.py [hunk 1]
```diff
@@ -1478,5 +1478,21 @@ def test_minimal(self):
...
+    
+    def test_snapshot(self):
+        sys.argv = ['qds.py', 'cluster', 'snapshot', '--label', '1234', '--parameters', ' {"s3_location":"myString", "backup_type":"full"}']
+        print_command()
+        Connection._api_call = Mock(return_value={})
+        qds.main()
+        Connection._api_call.assert_called_with('POST', 'clusters/1234/snapshot', {'parameters':  {'s3_location':'myString', 'backup_type':'full'}})
+
+    def test_restore_point(self):
+        sys.argv = ['qds.py', 'cluster', 'restore_point', '--label', '1234', '--parameters', ' {"s3_location":"myString", "backup_id":"abcd", "table_names":"tablename"}']
```
Comment (mindprince) [inline tests/test_cluster.py]: Wondering whether it is a good idea to ask users to write complicated string on the command-line.

File: qds_sdk/cluster.py [hunk 2]
```diff
@@ -353,6 +353,28 @@ def update(cls, cluster_id_label, cluster_info):
         return conn.put(cls.element_path(cluster_id_label), data=cluster_info)
 
     @classmethod
+    def _parse_cluster_manage_command(cls, args):
```
Comment (vrajat) [inline qds_sdk/cluster.py]: You can get rid of this function now.

File: qds_sdk/cluster.py [hunk 3]
```diff
@@ -353,6 +353,28 @@ def update(cls, cluster_id_label, cluster_info):
...
+        """
+        Parse command line arguments for cluster manage commands.
+        """
+        argparser = ArgumentParser(prog="cluster_manage_command")
+        group = argparser.add_mutually_exclusive_group(required=True)
+        group.add_argument("--id", dest="cluster_id",
+                          help="execute on cluster with this id")
+        group.add_argument("--label", dest="label",
```
Comment (vrajat) [inline qds_sdk/cluster.py]: Remember to add Required=True. Refer to https://docs.python.org/2/library/argparse.html

File: qds_sdk/cluster.py [hunk 4]
```diff
@@ -353,6 +353,28 @@ def update(cls, cluster_id_label, cluster_info):
...
+                          help="execute on cluster with this label")
+        argparser.add_argument("--parameters",
```
Comment (vrajat) [inline qds_sdk/cluster.py]: Not required

File: qds_sdk/cluster.py [hunk 5]
```diff
@@ -403,6 +425,73 @@ def delete(cls, cluster_id_label):
...
+    def _parse_snapshot(cls, args):
+        """
+        Parse command line arguments for snapshot command.
+        """
+        argparser = ArgumentParser(prog="snapshot")
+        
+        group = argparser.add_mutually_exclusive_group(required=True)
+        group.add_argument("--id", dest="cluster_id",
+                          help="execute on cluster with this id")
+        group.add_argument("--label", dest="label",
```
Comment (vrajat) [inline qds_sdk/cluster.py]: Remember to add required=True. https://docs.python.org/2/library/argparse.html

**Comment by adeshr** [comment]
Will reopen a PR wrt to unreleased branch.

**Pull request closed without merging**